packages/server/src/definitions/automations.ts

```diff
@@ -31,4 +34,6 @@ export interface AutomationContext {
     company?: string
   }
   loop?: { currentItem: any }
+  _stepIndex: number
+  _error: boolean
```

Comment:
`_stepIndex` tracks what numerical index to use next for step output (e.g. for `{{ steps.0.output }}`), and `_error` tracks whether a step has failed. This removes the need for us to loop over all of the output to check.